### ISSUE TITLE
Issue with Training on Custom Dataset  

### ISSUE THREAD

**Issue by rereset** [L-KID/Video-object-detection-by-location-anticipation#12]
Hi,
Thank you for providing this excellent repository. I am currently trying to train the model on my own dataset, which is in COCO format. However, I am encountering the following issue during training, and I am unsure how to resolve it:

  File "/root/autodl-tmp/VOD/pytorch_mask_rcnn/model/roi_heads.py", line 39, in fastrcnn_loss
    intermidiate_loss += F.smooth_l1_loss(regression[box_idx, label], intermidiate_targets[i], reduction='sum') / N
  File "/root/miniconda3/envs/efficientVOD/lib/python3.7/site-packages/torch/nn/functional.py", line 2606, in smooth_l1_loss
    expanded_input, expanded_target = torch.broadcast_tensors(input, target)
  File "/root/miniconda3/envs/efficientVOD/lib/python3.7/site-packages/torch/functional.py", line 71, in broadcast_tensors
    return _VF.broadcast_tensors(tensors)  # type: ignore
RuntimeError: The size of tensor a (6) must match the size of tensor b (3) at non-singleton dimension 0

Could you please advise if there are any specific requirements or potential reasons for this issue? For example, do single-category datasets require additional modifications, or could there be an issue with my data format?
Thank you for your help! Please let me know if more information is needed.

**Comment by Elstuhn** [comment]
Yeah I'm having this same issue, seems that sometimes the shape of intermidiate target and regression sometimes matches and sometimes does not match, let me know if you have solved this problem please! I'll be looking into it as well
